src/page-manager/wrapper.js

```diff
@@ -0,0 +1,173 @@
+/**                   _
+ *  _             _ _| |_
+ * | |           | |_   _|
+ * | |___  _   _ | | |_|
+ * | '_  \| | | || | | |
+ * | | | || |_| || | | |
+ * |_| |_|\___,_||_| |_|
+ *
+ * (c) Huli Inc
+ */
+
+/**
+ * Customize a route to be display inside a wrapper component like a toast
+ * or a dialog. This will also handle routing and dependencies loading
+ *
+ * @see  PageManager route option
+ */
+define([
+            'jquery',
+            'ju-shared/class'
+        ],
+        function(
+            $,
+            Class
+        ) {
+
+    'use strict';
+
+    var RouteWrapper = Class.extend({
+        /**
+         * Constructor
+         *
+         * @param  {object} controllerInfo  route defintion
```

Comment:
we should document which parts of the `controllerInfo` object are relevant to this  constructor